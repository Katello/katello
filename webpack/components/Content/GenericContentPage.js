import React from 'react';
import PropTypes from 'prop-types';
import { Grid, Col, Row, Form, FormGroup } from 'react-bootstrap';
import SearchBar from 'foremanReact/components/SearchBar';
import { getControllerSearchProps } from 'foremanReact/constants';
import ContentTable from './ContentTable';
import { useClearSearch } from '../extensions/SearchBar/SearchBarHooks';

const GenericContentPage = ({
  header, onSearch, bookmarkController,
  autocompleteEndpoint, autocompleteQueryParams,
  updateSearchQuery, initialInputValue,
  content, tableSchema, onPaginationChange,
}) => {
  const searchDataProp = {
    ...getControllerSearchProps(autocompleteEndpoint, `searchBar-content-page-${header}`, true, autocompleteQueryParams),
    controller: bookmarkController,
  };
  const searchBarKey = useClearSearch({ updateSearchQuery });
  return (
    <Grid bsClass="container-fluid">
      <Row>
        <Col sm={12}>
          <h1>{header}</h1>
        </Col>
      </Row>
      <Row>
        <Col sm={6}>
          <Form className="toolbar-pf-actions">
            <FormGroup className="toolbar-pf toolbar-pf-filter">
              <SearchBar
                key={searchBarKey}
                data={searchDataProp}
                onSearch={onSearch}
                onSearchChange={updateSearchQuery}
                initialInputValue={initialInputValue}
              />
            </FormGroup>
          </Form>
        </Col>
      </Row>
      <Row>
        <Col sm={12}>
          <ContentTable
            content={content}
            tableSchema={tableSchema}
            onPaginationChange={onPaginationChange}
          />
        </Col>
      </Row>
    </Grid>
  );
};

GenericContentPage.propTypes = {
  header: PropTypes.string.isRequired,
  content: PropTypes.shape({}).isRequired,
  tableSchema: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  onSearch: PropTypes.func.isRequired,
  updateSearchQuery: PropTypes.func.isRequired,
  initialInputValue: PropTypes.string.isRequired,
  onPaginationChange: PropTypes.func.isRequired,
  autocompleteEndpoint: PropTypes.string,
  autocompleteQueryParams: PropTypes.shape({}),
  bookmarkController: PropTypes.string,
};

GenericContentPage.defaultProps = {
  autocompleteEndpoint: undefined,
  autocompleteQueryParams: undefined,
  bookmarkController: undefined,
};

export default GenericContentPage;
