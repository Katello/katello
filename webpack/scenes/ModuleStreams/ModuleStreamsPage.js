import React, { Component } from 'react';
import PropTypes from 'prop-types';
import qs from 'query-string';
import { translate as __ } from 'foremanReact/common/I18n';
import { orgId } from '../../services/api';
import TableSchema from '../ModuleStreams/ModuleStreamsTableSchema';
import ContentPage from '../../components/Content/ContentPage';

class ModuleStreamsPage extends Component {
  constructor(props) {
    super(props);

    const queryParams = qs.parse(this.props.location.search);
    this.state = {
      searchQuery: queryParams.search || '',
    };
  }

  componentDidMount() {
    this.props.getModuleStreams({
      search: this.state.searchQuery,
    });
  }

  onPaginationChange = (pagination) => {
    this.props.getModuleStreams({
      ...pagination,
    });
  };

  onSearch = (search) => {
    this.props.getModuleStreams({ search });
  };

  getAutoCompleteParams = search => ({
    endpoint: '/module_streams/auto_complete_search',
    params: {
      organization_id: orgId(),
      search,
    },
  });

  updateSearchQuery = (searchQuery) => {
    this.setState({ searchQuery });
  };

  render() {
    const { moduleStreams } = this.props;
    return (
      <ContentPage
        header={__('Module Streams')}
        content={moduleStreams}
        tableSchema={TableSchema}
        onSearch={this.onSearch}
        getAutoCompleteParams={this.getAutoCompleteParams}
        updateSearchQuery={this.updateSearchQuery}
        initialInputValue={this.state.searchQuery}
        onPaginationChange={this.onPaginationChange}
      />
    );
  }
}

ModuleStreamsPage.propTypes = {
  location: PropTypes.shape({
    search: PropTypes.oneOfType([
      PropTypes.shape({}),
      PropTypes.string,
    ]),
  }),
  getModuleStreams: PropTypes.func.isRequired,
  moduleStreams: PropTypes.shape({}).isRequired,
};

ModuleStreamsPage.defaultProps = {
  location: { search: '' },
};

export default ModuleStreamsPage;
