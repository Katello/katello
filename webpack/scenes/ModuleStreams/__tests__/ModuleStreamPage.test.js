import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ModuleStreamsPage from '../ModuleStreamsPage';
import GenericContentPage from '../../../components/Content/GenericContentPage';

describe('Module streams page', () => {
  it('should render and contain appropiate components', async () => {
    const moduleStreams = {};
    const mockLocation = { search: '' };
    const getModuleStreams = () => {};

    const wrapper = shallow(<ModuleStreamsPage
      moduleStreams={moduleStreams}
      getModuleStreams={getModuleStreams}
      location={mockLocation}
    />);

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find(GenericContentPage)).toHaveLength(1);
  });
});

