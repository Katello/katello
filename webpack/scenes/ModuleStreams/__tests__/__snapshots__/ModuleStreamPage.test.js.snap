// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Module streams page should render and contain appropiate components 1`] = `
<Grid
  bsClass="container-fluid"
  componentClass="div"
  fluid={false}
>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      sm={12}
    >
      <h1>
        Module Streams
      </h1>
    </Col>
  </Row>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      sm={6}
    >
      <Form
        bsClass="form"
        className="toolbar-pf-actions"
        componentClass="form"
        horizontal={false}
        inline={false}
      >
        <FormGroup
          bsClass="form-group"
          className="toolbar-pf toolbar-pf-filter"
        >
          <Search
            getAutoCompleteParams={[Function]}
            initialInputValue=""
            onSearch={[Function]}
            updateSearchQuery={[Function]}
          />
        </FormGroup>
      </Form>
    </Col>
  </Row>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      sm={12}
    >
      <ModuleStreamsTable
        moduleStreams={Object {}}
        onPaginationChange={[Function]}
      />
    </Col>
  </Row>
</Grid>
`;
